src/index.ts

```diff
@@ -41,26 +41,39 @@ export const xplorersbot: HttpFunction = async (req, res) => {
     writeLog(log, message, LOG_METADATA);
 
     switch (req.body.type) {
-        case "url_verification":
-            res.status(200).send(req.body.challenge);
-            break;
         case "event_callback":
             const slackEvent = req?.body?.event;
+
+            if (slackEvent.bot_id) {
+                break;
+            }
+
             const isChannelOpenAI =
                 slackEvent?.channel ===
                 process.env.XPLORERS_OPENAI_SLACK_CHANNEL_ID;
-            const messageStartsWithHeyOpenAI = slackEvent?.text
-                .toLowerCase()
-                .startsWith("hey openai");
 
-            if (isChannelOpenAI && messageStartsWithHeyOpenAI) {
-                await createHttpTask(req.body);
+            // text could be in slackEvent.text or slackEvent.message.text
+            const message = slackEvent?.text ?? slackEvent?.message?.text;
+
+            if (isChannelOpenAI) {
```

Comment:
can be done in 1 level if `if(isChannelOpenAI && messageStartsWithHeyOpenAI)`... 

Up to you to decide :).

Reply:
Indeed, changed it to be 1 level 👍